extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/error/CancelationReason.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.msq.indexing.error;
+
+/**
+ * Enum denoting the reason for query cancelation.
+ */
+public enum CancelationReason
+{
+  /**
+   * Query was cancaled due to the task shutting down.
+   */
+  TASK_SHUTDOWN("Task shutdown"),
+  /**
+   * Query was cancaled due to a request from the user.
+   */
+  USER_REQUEST("User request"),
+  /**
+   * Query was canceled due to exceeding the configured query timeout.
+   */
+  QUERY_TIMEOUT("Query timeout"),
+  /**
+   * Query was canceled due to an unknown reason.
+   */
+  UNKNOWN("Unknown");
+
+  private final String reason;
+
+  CancelationReason(String reason)
```

Comment:
Some care needs to be taken here with serde. We need to be able to add new reasons in backwards-compatible ways so we can support rolling updates. The current approach will I think have a problem: when old code encounters a new reason, it will fail to deserialize with some error about there being no such enum value.

A way to solve this is to add a `@JsonCreator` static creator method that returns `UNKNOWN` when it encounters a name that it doesn't recognize.

Reply:
Thanks for catching this, changed this and added a case in `CanceledFaultTest` for new reasons